### PR TITLE
feat: set spending cap screen

### LINKED ISSUE #20335: Trigger Swap Screen
image

https://www.figma.com/design/AD2JSKg0I8dZcylyiGa62O/Swap-for-Mobile?node-id=48-133021&m=dev

### LINKED ISSUE #20336: Trigger Set spending cap Drawer
https://www.figma.com/design/AD2JSKg0I8dZcylyiGa62O/Swap-for-Mobile?node-id=48-98051&m=dev

image

### PR DESCRIPTION
**Pull request by briansztamfater**
fixes #20336  

### Summary

This PR implements Set Spending Cap screen, no backend integration yet and some data is hardcoded for now, but UI should be functional and most of the logic should be in place

#### Platforms

- Android
- iOS

##### Functional

- wallet / transactions

### Steps to test

- Open Status
- Log in
- Open wallet
- Select an account
- Tap on Swap
- Select an asset in asset to pay screen
- Tap on Set spending cap button to navigate to Set spending cap screen
- Verify UI looks as expected

status: ready

### CODE REVIEW

**Comment by ajayesivan** [comment]
Error after slide to swap:
https://github.com/user-attachments/assets/80c6bbe5-fd1e-423e-9dbc-6548f5eddbb4

**Review by ajayesivan — APPROVED** [review]
Amazing work 🎉

File: src/status_im/contexts/wallet/swap/set_spending_cap/view.cljs [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+        native-currency-symbol  (get-in swap-proposal [:from :native-currency-symbol])
+        account                 (rf/sub [:wallet/current-viewing-account])
+        account-color           (:color account)
+        provider                (first providers)]
+    [rn/view {:style style/container}
+     [floating-button-page/view
+      {:footer-container-padding 0
+       :header                   [quo/page-nav
+                                  {:icon-name           :i/close
+                                   :on-press            on-close-action
```
Comment (mohsen-ghafouri) [inline src/status_im/contexts/wallet/swap/set_spending_cap/view.cljs]:
```suggestion
                                   :on-press            navigation/navigate-back
```
from this package          [utils.navigation :as navigation]

**Review by mohsen-ghafouri — APPROVED** [review]
LGTM, just one small comment :)

File: src/status_im/contexts/wallet/swap/set_spending_cap/view.cljs (hunk 1 above)
Comment (briansztamfater) [reply src/status_im/contexts/wallet/swap/set_spending_cap/view.cljs]: Fixed!

**Comment by briansztamfater** [comment]
> Error after slide to swap: https://github.com/user-attachments/assets/80c6bbe5-fd1e-423e-9dbc-6548f5eddbb4

@ajayesivan Fixed the issue and added a temporary alert to prevent the error.

**Review by smohamedjavid — APPROVED** [review]
LGTM 🚀 🙌

**Comment by mariia-skrypnyk** [comment]
Hi @briansztamfater !

 Is it correct that this PR should skip manual QA part?

**Comment by briansztamfater** [comment]
> Hi @briansztamfater !
> 
>  Is it correct that this PR should skip manual QA part?

HI @mariia-skrypnyk! QA is not strictly required in this case because this PR only includes screen UI with hardcoded data, so it is better to finish integration before involving QA.

**Comment by briansztamfater** [comment]
@status-im/mobile-qa  Can you check failed e2e tests please? Thanks in advance!

**Comment by pavloburykh** [comment]
@briansztamfater thanks! PR is ready for merge.

**Pull request merged**